luatest/treegen.lua

```diff
@@ -0,0 +1,210 @@
+--- Working tree generator.
+--
+-- Generates a tree of Lua files using provided templates and
+-- filenames.
+--
+-- @usage
+--
+-- local t = require('luatest')
+-- local treegen = require('test.treegen')
+--
+-- local g = t.group()
+--
+-- local SCRIPT_TEMPLATE = [[
+--     <...>
+-- ]]
+--
+-- g.before_all(function(g)
+--     treegen.init(g)
+--     treegen.add_template(g, '^.*$', SCRIPT_TEMPLATE)
+-- end)
+--
+-- g.after_all(function(g)
+--     treegen.clean(g)
+-- end)
```

Comment:
We don't need explicit `treegen.init` and `treegen.clean` calls since they're in the predefined hooks now.

Reply:
Removed.

Reply:
Let's either use or remove `SCRIPT_TEMPLATE` in this example.

Reply:
Removed.

Reply:
Now the example attempts to create files without matching templates -- I guess it leads to attempt to call `:gsub()` on a `nil` template value.

Please, provide any working example or drop it.

BTW, `require('test.treegen')` should be `require('luatest.treegen')`.

Reply:
Yep, fixed it. Thanks!

Reply:
:rocket:  fixed.

Reply:
OK, it should work.

However, if a template is added in the test case itself, it may affect other test cases in the same group in case when the test fails. It is why I [asked](https://github.com/tarantool/luatest/pull/371#discussion_r1640113121) to do so in before/after test hooks.

If a template is written for all the test cases of the given test, {before,after}_all can be used instead.

Reply:
I'll fix it as a follow up. Closing the discussion.